worker/sink_handler.go

```diff
@@ -86,10 +86,18 @@ func newKafkaSink(config *z.SuperFlag) (Sink, error) {
 	saramaConf.Producer.Return.Successes = true
 	saramaConf.Producer.Return.Errors = true
 
-	if config.GetPath("ca-cert") != "" {
-		tlsCfg := &tls.Config{
-			MinVersion: tls.VersionTLS12,
+	if config.GetBool("tls") && config.GetPath("ca-cert") == "" {
+		tlsCfg := x.TLSBaseConfig()
```

Comment:
There's quite a lot of this duplicated in the `elif` test below. Not a deal-breaker, but definitely could be cleaned up.

Reply:
a few lines yeah. I tried putting this code inside the TLSBaseConfig function but realized that it can error out while reading system certs. We may not want that in certain cases.